aya/src/obj/mod.rs

```diff
@@ -342,16 +375,18 @@ impl Object {
 
         match section.kind {
             BpfSectionKind::Data => {
-                self.maps
-                    .insert(section.name.to_string(), parse_map(&section, section.name)?);
+                self.maps.insert(
```

Comment:
I think you should revert parse_map as it was before, and instead do something like:

```
let map = parse_map(&section, name)?;
patch_data_map(map.data, globals, self.symbols_by_index)?;
self.maps.insert(name.to_string(), map);
```

Reply:
done